File: JAVA/TomadasTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */
package JAVA;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class TomadasTest {

    @Test
    public void testTomadas() throws Exception {
      //Test 1
      int want = 8;
      int got = Tomadas.maxDevices(2, 4, 3, 2);
      assertEquals(want, got);
      //Test 2
      want = 21;
      got = Tomadas.maxDevices(6, 6, 6, 6);
      assertEquals(want, got);
      //Test 3
      want = 5;
      got = Tomadas.maxDevices(2, 2, 2, 2);
      assertEquals(want, got);
    }
}

File: PHP/TomadasTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("Tomadas.php");

class TomadasTest extends TestCase {

    public function testTomadas() {
        $obj = new Tomadas();
        //Test 1
        $want = 8;
        $got = $obj->maxDevices(2, 4, 3, 2);
        $this->assertEquals($want, $got);
        //Test 2
        $want = 21;
        $got = $obj->maxDevices(6, 6, 6, 6);
        $this->assertEquals($want, $got);
        //Test 3
        $want = 5;
        $got = $obj->maxDevices(2, 2, 2, 2);
        $this->assertEquals($want, $got);
    }

}
